app/router/v2/tags.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from app.database.beatmap import Beatmap
+from app.database.beatmap_tags import BeatmapTagVote
+from app.dependencies.database import get_db
+from app.dependencies.user import get_current_user
+from app.models.tags import BeatmapTags, get_all_tags, get_tag_by_id
+
+from .router import router
+
+from fastapi import Depends, HTTPException
+from pydantic import BaseModel
+from sqlmodel import select
+from sqlmodel.ext.asyncio.session import AsyncSession
+
+
+class APITagCollection(BaseModel):
+    tags: list[BeatmapTags]
+
+
+@router.get(
+    "/tags",
+    tags=["用户标签"],
+    response_model=APITagCollection,
+    name="获取所有标签",
+    description="获取所有可用的谱面标签。",
+)
+async def router_get_all_tags():
+    """
+    获取所有可用标签。
+    返回系统中所有可用的谱面标签列表。
+    """
+    return APITagCollection(tags=get_all_tags())
+
+
+@router.put(
+    "/beatmaps/{beatmap_id}/tags/{tag_id}",
+    tags=["用户标签"],
+    status_code=201,
```

Comment:
```suggestion
    status_code=204,
```

参考 https://github.com/ppy/osu-web/blob/0b569ed14ab6dce57852625727dbd7768e7be2f5/app/Http/Controllers/BeatmapTagsController.php